Updated 0.9.6

dgarcia360 opened the pull request:
Applied address changes from https://github.com/nemtech/catbuffer/commit/389c96e49961354767e8b5688a1e52cacd14e0f9

@Vektrat please let me know if the implementation matches the spec once you apply the changes on the REST side.

Vektrat commented:
hmm... catbuffer changes have nothing to do with API i believe, you can take those as hints for stuff that has changed in mongodb most of the times, but it's not deterministic

we are following [core changes](https://github.com/nemtech/catapult-server/commits/master) to [update the schemas](https://github.com/nemtech/catapult-rest/pull/348)

dgarcia360 commented:
It's not deterministic but should be since catbuffer has as well the state schemas represented. Of course, there can be inconsistencies, but in this case, I think it should be reported to the catbuffer repo + OpenAPI. If you find something different from the changes proposed, please post it here. Thanks!

dgarcia360 commented:
It's not deterministic but should be since catbuffer has as well the state schemas represented. Of course, there can be inconsistencies, but in this case, I think it should be reported to the catbuffer repo + OpenAPI. If you find something different from the changes proposed, please post it here.

rg911 commented on spec/schemas/AddressBase32.yml:
```diff
@@ -1,3 +1,3 @@
 type: string
 description: Address encoded using a 32-character set.
```
39 instead of 32

dgarcia360 replied:
Looks correct. It' encoded using [32-charater set](https://en.wikipedia.org/wiki/Base32) - comprised of comprising the twenty-six upper-case letters A–Z, and the digits 2–7 - although the length is 39.

rg911 commented on spec/plugins/metadata/schemas/AccountMetadataTransactionBodyDTO.yml:
```diff
@@ -1,13 +1,13 @@
-  targetPublicKey:
-    $ref: "../../../schemas/PublicKey.yml"
+  targetAddress:
+    $ref: "../../../schemas/Address.yml"
```
Should be `UnresolvedAddress`

dgarcia360 replied:
good catch, solved

rg911 commented on spec/plugins/metadata/schemas/MetadataEntryDTO.yml:
```diff
@@ -1,18 +1,18 @@
-  targetPublicKey:
-    $ref: "../../../schemas/PublicKey.yml"
+  sourceAddress:
+    $ref: "../../../schemas/Address.yml"
```
UnresolvedAddress

rg911 commented on spec/plugins/metadata/schemas/MosaicMetadataTransactionBodyDTO.yml:
```diff
@@ -1,14 +1,14 @@
-  targetPublicKey:
-    $ref: "../../../schemas/PublicKey.yml"
+  targetAddress:
+    $ref: "../../../schemas/Address.yml"
```
UnresolvedAddress

rg911 commented on spec/plugins/metadata/schemas/NamespaceMetadataTransactionBodyDTO.yml:
```diff
@@ -1,14 +1,14 @@
-  targetPublicKey:
-    $ref: "../../../schemas/PublicKey.yml"
+  targetAddress:
+    $ref: "../../../schemas/Address.yml"
```
UnresolveAddress

rg911 commented on spec/plugins/multisig/schemas/MultisigAccountModificationTransactionBodyDTO.yml:
```diff
@@ -17,13 +17,13 @@ properties:
     items:
-      $ref: "../../../schemas/PublicKey.yml"
-  publicKeyDeletions:
+      $ref: "../../../schemas/Address.yml"
```
UnresolvedAddress

Vektrat commented:
> Metadata endpoints seem not correct as still using sender's PublicKey, should it be changed to Address instead?
> 
> @Vektrat

indeed metadata is now using `sourceAddress` and `targetAddress`

fboucquez commented:
Would it be possible to clarify if the stored entities from transactions have aliases resolved or not? Are we expecting alises when loading metadata, namespaces, restrictions etc, or resolved real addresses? Knowing what to expect now will save us a lot of time. If not we would need to try and error each individual case against a server (which we don't have yet) and update the models/services adding alias resolution when required.

@Jaguar0625 @Wayonb @gimer ?

@Vektrat what's your experience from testing rest?

Vektrat commented:
> Would it be possible to clarify if the stored entities from transactions have aliases resolved or not? Are we expecting alises when loading metadata, namespaces, restrictions etc, or resolved real addresses? Knowing what to expect now will save us a lot of time. If not we would need to try and error each individual case against a server (which we don't have yet) and update the models/services adding alias resolution when required.
> 
> @Jaguar0625 @Wayonb @gimer ?
> 
> @Vektrat what's your experience from testing rest?

aren't those unresolved? can't this be checked in catbuffer schemas? this is of no importance to rest so we haven't really paid it much attention

fboucquez commented:
I actually missed that the catbuffer schemas include the entity schemas, not only the transactions. Now it's crystal clear.  Spec seems fine for now. Thanks!!!

dgarcia360 commented:
Added the new response property ``size`` in TransactionDTO and BlockDTO.

dgarcia360 commented:
> Missing status 0x804C0003

Added https://github.com/nemtech/symbol-openapi/pull/143/commits/bf6f567d1781631cfce723b2c50b04ece1903dbf

> Metadata get by sender is still using senderPublicKey, should be changed to address

Fixed https://github.com/nemtech/symbol-openapi/pull/156

* Updated network properties: https://github.com/nemtech/symbol-openapi/pull/157

dgarcia360 commented:
@Vektrat @martinayora please review
